### ISSUE TITLE
🐛 `lint/suspicious/noRedeclare` false positive

### ISSUE THREAD

**Issue by nikeee** [biomejs/biome#591]
### Environment information

```block
CLI:
  Version:                      1.3.1
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v21.0.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 false

Workspace:
  Open Documents:               0
```

### What happened?

```
interface A {
	get<T>(f: T): T;
	post<T>(g: T): T;
}
```

Yields error:
```
  ✖ Shouldn't redeclare 'T'. Consider to delete it or rename it.
  
    1 │ interface A {
    2 │ 	get<T>(f: T): T;
  > 3 │ 	post<T>(g: T): T;
      │ 	    ^
    4 │ }
    5 │ 
  
  ℹ 'T' is defined here:
  
    1 │ interface A {
  > 2 │ 	get<T>(f: T): T;
      │ 	   ^
    3 │ 	post<T>(g: T): T;
    4 │ }
```

### Expected result

It should not complain.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct
